### PR TITLE
Seq: rename {push,pop}->{push,pop}_back, add {push,pop}_front

### PR DESCRIPTION
**Pull request by Armael**

### CODE REVIEW

File: creusot-contracts/src/logic/seq.rs [hunk 1]
```diff
@@ -134,11 +134,19 @@ impl<T: ?Sized> Seq<T> {
         absurd
     }
 
+    #[trusted]
+    #[logic]
+    #[open(self)]
+    #[creusot::builtins = "prelude.prelude.Seq.cons_"]
```
Comment (arnaudgolfouse) [inline creusot-contracts/src/logic/seq.rs:140]: this uses "prelude.prelude.Seq.cons_", but push_back uses "seq.Seq.snoc", is there a reason for this difference?

Comment (Armael) [reply creusot-contracts/src/logic/seq.rs:140]: `seq.Seq.snoc` is from the why3 stdlib. Ideally we would also use `seq.Seq.cons` (also from the why3 stdlib), but the order of arguments is different from the one we want! (`self` has to be the first argument, right?)
Hence the `cons_` wrapper to swap the arguments.

Comment (xldenis) [reply creusot-contracts/src/logic/seq.rs:140]: I guess this is moving the wrapper down one level.

Comment (xldenis) [reply creusot-contracts/src/logic/seq.rs:140]: My recollection was that Seq had an internal wrapper that flipped the arguments

Comment (Armael) [reply creusot-contracts/src/logic/seq.rs:140]: I couldn't see such a wrapper, but maybe I missed it?

Comment (xldenis) [reply creusot-contracts/src/logic/seq.rs:140]: https://github.com/creusot-rs/creusot/blob/master/creusot-contracts/src/logic/seq.rs#L48 ?

Comment (xldenis) [reply creusot-contracts/src/logic/seq.rs:140]: Ah i guess i only did it for sets: https://github.com/creusot-rs/creusot/blob/master/creusot-contracts/src/logic/fset.rs#L50-L52

Comment (xldenis) [reply creusot-contracts/src/logic/seq.rs:140]: we're not 100% consistent either way, I have a slight preference for the rust wrapper approach but i wouldn't consider it blocking.

Comment (Armael) [reply creusot-contracts/src/logic/seq.rs]: I agree that it is better. I pushed a new version with a rust wrapper instead of one in why3

**Comment by Armael** [comment]
I keep getting owned by tests that replay locally but not in CI… I have the suspicion that the new "Creusot" strategy in why3ide is generating more fragile sessions than the others because of the short initial timeout for z3.

**Comment by xldenis** [comment]
> I keep getting owned by tests that replay locally but not in CI… I have the suspicion that the new "Creusot" strategy in why3ide is generating more fragile sessions than the others because of the short initial timeout for z3.

Oh that's a good point. It really reinforces that we should make steps work for our sessions. This would ensure CI is as reproducible as local.

**Pull request merged**